activerecord/lib/active_record/store.rb

```diff
@@ -81,19 +86,26 @@ class << self
     module ClassMethods
       def store(store_attribute, options = {})
         serialize store_attribute, IndifferentCoder.new(store_attribute, options[:coder])
-        store_accessor(store_attribute, options[:accessors]) if options.has_key? :accessors
+        store_accessor(store_attribute, options[:accessors], prefix: options[:prefix]) if options.has_key? :accessors
       end
 
-      def store_accessor(store_attribute, *keys)
+      def store_accessor(store_attribute, *keys, prefix: nil)
         keys = keys.flatten
 
+        accessor_prefix =
+          if prefix
+            "#{prefix == true ? store_attribute : prefix}_"
+          else
+            ""
+          end
```

Comment:
I think this might look better as:

``` ruby
accessor_prefix =
  case prefix
  when String, Symbol
    "#{prefix}_"
  when TrueClass
    "#{store_attribute}_"
  else
    ""
  end
```

it makes the intent clearer than hiding inside a ternary in a string interpolation.